data-reporting-service/src/main/resources/db/sp-provider-001.sql

```diff
@@ -1,133 +1,126 @@
-                                              au.last_chg_time                                                               as ADD_USER_CHG_TIME,
-                                              CAST((RTRIM(au.user_first_nm) + ', ' + RTRIM(au.user_last_nm)) as varchar(50)) AS patient_added_by
-                                       FROM NBS_ODSE.dbo.Auth_user au WITH (NOLOCK)
-                                       WHERE p.ADD_USER_ID = au.NEDSS_ENTRY_ID
-                                       FOR json path, INCLUDE_NULL_VALUES) AS authadd) AS authadd,
-                              --auth user change
-                              (SELECT (SELECT p.last_chg_user_id,
-                                              au.add_time,
-                                              au.last_chg_time                                                               AS LAST_CHG_USER_TIME,
-                                              CAST((RTRIM(au.user_first_nm) + ', ' + RTRIM(au.user_last_nm)) as varchar(50)) AS patient_last_updated_by
-                                       FROM NBS_ODSE.dbo.Auth_user au WITH (NOLOCK)
-                                       WHERE p.LAST_CHG_USER_ID = au.NEDSS_ENTRY_ID
-                                       FOR json path, INCLUDE_NULL_VALUES) AS authchg) AS authchg) AS nested
-    WHERE p.person_uid in (SELECT value FROM STRING_SPLIT(@user_id_list, ','))
-      AND p.cd = 'PRV'
 
+    BEGIN TRY
+        SELECT p.person_uid,
+               p.person_parent_uid,
+               p.description,
+               p.add_time,
+               p.first_nm,
+               p.middle_nm,
+               p.last_nm,
+               p.nm_suffix,
+               p.cd,
+               p.electronic_ind,
+               p.last_chg_time,
+               p.record_status_cd,
+               p.record_status_time,
+               p.status_cd,
+               p.status_time,
+               p.local_id,
+               p.version_ctrl_nbr,
+               p.edx_ind,
+               p.dedup_match_ind,
+               p.add_user_id,
+               case when p.add_user_id > 0 then (select * from dbo.fn_get_user_name(p.add_user_id))
+                   end as add_user_name,
+               p.last_chg_user_id ,
+               case when p.last_chg_user_id > 0 then (select * from dbo.fn_get_user_name(p.last_chg_user_id))
+                   end as last_chg_user_name,
+               nested.name      AS 'provider_name',
+               nested.address   AS 'provider_address',
+               nested.phone     AS 'provider_telephone',
+               nested.email     AS 'provider_email',
+               nested.entity_id AS 'provider_entity'
+        FROM nbs_odse.dbo.Person p WITH (NOLOCK)
+                 OUTER apply (SELECT *
+                              FROM
+                                  -- address
+                                  (SELECT (SELECT pl.postal_locator_uid,
+                                                  elp.cd elp_cd,
```

Comment:
If possible, could we update this with the latest provider sproc?

Reply:
Sure.